tiledb/bioimg/visualization/helpers.py

```diff
@@ -0,0 +1,65 @@
+import json
+from dataclasses import dataclass
+from typing import List, Optional, Sequence
+
+import tiledb
+
+from ..converters.axes import Axes
+
+
+@dataclass(frozen=True)
+class ZoomLevelRecord:
+    zoom_level: int
+    image_level: int
+    downsample: int
+    width: int
+    height: int
+    namespace: str
+    array_uri: str
+    axes: str
+
+
+def calculate_metadata(
+    namespace: str, group_uri: str, config: Optional[tiledb.Config] = None
+) -> Sequence[ZoomLevelRecord]:
```

Comment:
`Metadata` is too overloaded and generic term in this project. I'd rename it to (say) `iter_zoom_level_records` and change the return type to `Iterator[ZoomLevelRecord]`.

Reply:
It should be possible to wrap a generator into a `list()` before passing it, no?